### PR TITLE
[LIB-352] quickfix - update for object with files

### PR DESCRIPTION
**Pull request by Zhebr**
tests added plus corrected yesterdays stuff

### CODE REVIEW

File: tests/test_connection.py [hunk 1]
```diff
@@ -213,6 +213,31 @@ def test_invalid_method_name(self):
         with self.assertRaises(SyncanoValueError):
             self.connection.make_request('INVALID', 'test')
 
+    @staticmethod
+    def _create_test_file():
```
Comment (opalczynski) [inline tests/test_connection.py]: Maybe we should use here tempfile?

https://docs.python.org/2/library/tempfile.html

File: syncano/connection.py [hunk 2]
```diff
@@ -236,9 +236,12 @@ def make_request(self, method_name, path, **kwargs):
...
         data = kwargs.get('data', {})
 
-        files = {k: v for k, v in data.iteritems()
-                 if hasattr(v, 'read')}
-        map(data.pop, files.keys())
+        if method_name == 'POST':
+            files = data.pop('files', {})
+        else:
+            files = {k: v for k, v in data.iteritems()
+                     if hasattr(v, 'read')}
```
Comment (opalczynski) [inline syncano/connection.py]: Maybe here checks for read and write? It's always less possible to have interface with read and write than only read :) Dunno :)

Comment (Zhebr) [inline syncano/connection.py]: OK

File: tests/test_connection.py (hunk 1 above)
Comment (23doors) [inline tests/test_connection.py]: You could also use StringIO: https://docs.python.org/2/library/stringio.html
But I guess real or temp file creates a more "real" environment for tests even though StringIO has pretty much the same interface.

Comment (Zhebr) [inline tests/test_connection.py]: it's neat but will raise IOError for ```patch``` method - need to mock it

Comment (23doors) [inline tests/test_connection.py]: Why? tempfile is just a regular file that can be cleaned up when you're done.

**Pull request merged**